storage/filestore.go

```diff
@@ -61,6 +67,71 @@ type FilesystemStore struct {
 	perms   os.FileMode
 }
 
+func (f *FilesystemStore) moveKeyTo0Dot4Location(file string) {
+	keyID := filepath.Base(file)
+	fileDir := filepath.Dir(file)
+	d, _ := f.Get(file)
+	block, _ := pem.Decode(d)
+	if block == nil {
+		logrus.Warn("Key data for ", file, " may have been tampered with/ is invalid. The key has not been migrated and may not be available")
+		return
+	}
+	fileDir = strings.TrimPrefix(fileDir, notary.RootKeysSubdir)
+	fileDir = strings.TrimPrefix(fileDir, notary.NonRootKeysSubdir)
+	if fileDir != "" {
+		block.Headers["gun"] = fileDir[1:]
+	}
+	if strings.Contains(keyID, "_") {
+		role := strings.Split(keyID, "_")[1]
+		keyID = strings.TrimSuffix(keyID, "_"+role)
+		block.Headers["role"] = role
+	}
+	var keyPEM bytes.Buffer
+	// since block came from decoding the PEM bytes in the first place, and all we're doing is adding some headers we ignore the possibility of an error while encoding the block
+	pem.Encode(&keyPEM, block)
+	f.Set(keyID, keyPEM.Bytes())
+}
+
+func (f *FilesystemStore) migrateTo0Dot4() {
+	if _, err := os.Stat(filepath.Join(f.Location(), notary.RootKeysSubdir)); !os.IsNotExist(err) {
+		// root_keys exists, migrate things from it
+		listOnlyRootKeysDirStore, _ := NewFileStore(filepath.Join(f.baseDir, notary.RootKeysSubdir), f.ext, notary.PrivKeyPerms)
+		for _, file := range listOnlyRootKeysDirStore.ListFiles() {
+			f.moveKeyTo0Dot4Location(filepath.Join(notary.RootKeysSubdir, file))
+		}
+		// delete the old directory
+		rootKeysSubDir := filepath.Join(f.Location(), notary.RootKeysSubdir)
+		if rootKeysSubDir == "" || rootKeysSubDir == "/" {
```

Comment:
I think the intention of this check is to make sure that we don't accidentally delete a users root dir (very rare case but still)
I'll clean up the variables as you recommended